src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/RetargetCustomAttributes.cs

```diff
@@ -321,5 +321,77 @@ public class C
             Assert.Equal("int[]", named["F"].Type.ToDisplayString());
             Assert.Throws<InvalidOperationException>(() => named["F"].Value);
         }
+
+        [Fact]
+        [WorkItem(65048, "https://github.com/dotnet/roslyn/issues/65048")]
+        public void MissingAttributeType()
+        {
+            var source1 = @"
+using System;
+
+public class A : Attribute
+{
+}
+";
+
+            var comp1 = CreateCompilation(source1, options: TestOptions.DebugDll);
+
+            var source2 = @"
+[A]
+public class C1
+{}
+";
+
+            var comp2 = CreateCompilation(source2, references: new[] { comp1.ToMetadataReference() }, options: TestOptions.DebugDll);
+            var comp3 = CreateCompilation("", references: new[] { comp2.ToMetadataReference() }, options: TestOptions.DebugDll);
```

Comment:
For my own edification: is the fact that `A` is missing here the reason we get into a retargeting scenario at all?

Reply:
> For my own edification: is the fact that ```A``` is missing here the reason we get into a retargeting scenario at all?

To be precise, the fact that the reference for `comp1` is not provided is the reason for retargeting.